VectorWidthB support, VectorWidth + non SourceSwap support, small bug fix for ClusterLocalRead

nakajee opened the pull request:
- supported VectorWidthB (added a new parameter VectorWidthB)
- supported VectorWidth + non SourceSwap
- fixed innacurate vgpr allocation for ClusterLocalRead
- added test cases for VectorWidthB, VectorWidth + non SourceSwap

AlexBrownAMD commented on Tensile/Common.py:
```diff
@@ -1223,9 +1223,9 @@ def getArchitectureName(gfxName):
-    # For MFMA non SourceSwap: this parameter didn't take effect
-    # For MFMA SourceSwap: this parameter only take effect on A buffer for now
     "VectorWidth":                [ -1, 1, 2, 3, 4, 6, 8, 16 ],
+    # VectorWidth for B (MatrixInstruction only)
+    "VectorWidthB":               [ -1, 1, 2, 4, 8, 16 ],
```
Does this refer to VectorWidth for B matrix or for block dimension of matrix instruction?

AlexBrownAMD replied:
Looks like matrix B, so "VectorWidth" now becomes "VectorWidthA"?

nakajee replied:
> Looks like matrix B, so "VectorWidth" now becomes "VectorWidthA"?

Internally, I added kernel["VectorWidthA"] (the value is same as "VectorWidth" ).
It is very difficult to change "VectorWidth" to "VectorWidthA".
I need to change all existing test cases and logic yaml files....

nakajee replied:
As I put a comment here, VectorWidthB is for MatrixInstruction only.
We still use VectorWidth for non MI part.
Of course we can change them to VectorWidthA. This is just to make sure we did not change the logic for non MI part.

nakajee commented:
Just added small fixes
- fixed mismatch issue with LdsBlockSizePerPad + MT1(or 0) not power of 2
- removed VectorWidth conversion for non SourceSwap in LdsPad part
- fixed a reject comment

Also, merged the latest develop change. Hopefully, CI fails will be fixed....

nakajee commented:
gfx908 extended case failed due to xfail....
It looks like 1sum_simple and 1sum_gsu_simple fail only on gfx90a...
(gfx90a extended passed with xfail)
